synapseclient/models/mixins/table_components.py

```diff
@@ -0,0 +1,1831 @@
+import asyncio
+import hashlib
+import json
+import logging
+import os
+import re
+import tempfile
+import time
+import uuid
+from dataclasses import field
+from typing import Dict, List, Optional, Tuple, Union
+from collections import OrderedDict
+
+
+from tqdm import tqdm
+from tqdm.contrib.logging import logging_redirect_tqdm
+
+from synapseclient import Synapse
+from synapseclient.api import (
+    ViewTypeMask,
+    delete_entity,
+    get_columns,
+    get_default_columns,
+    get_from_entity_factory,
+    post_columns,
+    post_entity_bundle2_create,
+    put_entity_id_bundle2,
+)
+from synapseclient.core.constants import concrete_types
+from synapseclient.core.exceptions import SynapseTimeoutError
+from synapseclient.core.utils import (
+    MB,
+    delete_none_keys,
+    log_dataclass_diff,
+    merge_dataclass_entities,
+)
+from synapseclient.models.services.search import get_id
+
+from synapseclient.core.async_utils import async_to_sync, otel_trace_method
+from synapseclient.models.services.storable_entity_components import (
+    FailureStrategy,
+    store_entity_components,
+)
+from synapseclient.models.mixins.table_operator import (
+    Column,
+    ColumnType,
+    TableSchemaChangeRequest,
+    ColumnChange,
+    TableUpdateTransaction,
+    DATA_FRAME_TYPE,
+    PartialRowSet,
+    PartialRow,
+    AppendableRowSetRequest,
+    CLASSES_THAT_CONTAIN_ROW_ETAG,
+    csv_to_pandas_df,
+    test_import_pandas,
+    _convert_pandas_row_to_python_types,
+    QueryResultBundle,
+    SumFileSizes,
+    SERIES_TYPE,
+)
+
+
+class TableBase:
+    """Mixin that extends the functionality of any `table` like entities in Synapse
+    to perform a number of operations on the entity such as getting, deleting, or
+    updating columns, querying for data, and more."""
+
+    id: None = None
+    name: None = None
+    parent_id: None = None
+    activity: None = None
+    version_number: None = None
+    _last_persistent_instance: None = None
+    _columns_to_delete: Optional[Dict] = None
+
+    def _set_last_persistent_instance(self) -> None:
+        """Used to satisfy the usage in this mixin from the parent class."""
+
+    def to_synapse_request(self) -> Dict:
+        """Used to satisfy the usage in this mixin from the parent class."""
+
+    def fill_from_dict(self, entity: Dict, set_annotations: bool) -> None:
+        """Used to satisfy the usage in this mixin from the parent class."""
+
+    @property
+    def has_changed(self) -> bool:
+        """Used to satisfy the usage in this mixin from the parent class."""
+
+    @property
+    def has_columns_changed(self) -> bool:
+        """Used to satisfy the usage in this mixin from the parent class."""
+
+
+class ViewBase(TableBase):
```

Comment:
Since this is introducing dataclass fields (`from dataclasses import field`), does this need the `@dataclass` decorator?